tsdb/index/index.go

```diff
@@ -270,37 +331,51 @@ func (w *Writer) ensureStage(s indexWriterStage) error {
 	if w.stage == s {
 		return nil
 	}
+	if w.stage < s-1 {
+		// A stage has been skipped.
+		if err := w.ensureStage(s - 1); err != nil {
```

Comment:
Am I right that no symbols, no series is the only case when this happens? 

I know we check later:

```
// Populated block is empty, so exit early.
	if meta.Stats.NumSamples == 0 {
		return nil
	}

```

Curious what's changed here in this area: e.g why suddenly we have to add this? Was it a bug that was there all the time? (empty index was not a valid index before?)

Reply:
I've had to do a few workarounds to the tests as I've been making these changes to prevent this. There's one unittest specifically testing for this.

Reply:
ok